### PR TITLE
Fix missing breaks in switches

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: inc/commondbtm.class.php [hunk 1]
```diff
@@ -3844,6 +3845,7 @@ function filterValues($display = true) {
                      if (!class_exists($value)) {
                         $unset = true;
                      }
+                     //go threw plugins
```
Comment (trasher) [inline inc/commondbtm.class.php:3848]: Not sure.

File: inc/dashboard/provider.class.php [hunk 2]
```diff
@@ -338,41 +338,47 @@ static function nbTicketsGeneric(
             $params['icon']  = Ticket::getIcon();
             $params['label'] = __("Incoming tickets");
             $skip = true;
+            //no break
```
Comment (trasher) [inline inc/dashboard/provider.class.php:341]: Do not know why

File: inc/search.class.php [hunk 3]
```diff
@@ -6124,6 +6126,7 @@ static function giveItem($itemtype, $ID, array $data, $meta = 0,
                                                                    'display' => false]));
                   return $out;
                }
+               break;
```
Comment (trasher) [inline inc/search.class.php:6129]: Not sure

File: inc/toolbox.class.php [hunk 4]
```diff
@@ -1503,6 +1503,7 @@ static function manageRedirect($where) {
                      // no break
                      case "tracking" :
                         $data[0] = "Ticket";
+                                //var defined, use default case
```
Comment (trasher) [inline inc/toolbox.class.php:1506]:
```suggestion
                        //var defined, use default case
```

File: inc/transfer.class.php [hunk 5]
```diff
@@ -3410,6 +3410,7 @@ function transferDevices($itemtype, $ID, $newID) {
                   ]
                );
             }
+                break;
```
Comment (trasher) [inline inc/transfer.class.php:3413]:
```suggestion
            break;
```

File: inc/dashboard/provider.class.php (hunk 2 above)
Comment (orthagh) [reply inc/dashboard/provider.class.php:341]: avoid repetition mainly, maybe a bad design choice on my side.
You can rewrite it, i'm not against

Comment (trasher) [reply inc/dashboard/provider.class.php:341]: no, ok for me, I better understand now

File: inc/commondbtm.class.php (hunk 1 above)
Comment (cedric-anne) [reply inc/commondbtm.class.php:3848]: IMHO, we should break here.
```suggestion
                     break;
```

I guess lack of break here is due to a bad code design when this case was added.
https://github.com/glpi-project/glpi/commit/f4f1b0425e3fb9e59e26c90825944184a96308d3#diff-bf93ba44eb9d93a8ce19e63a5b53377d4563ea9df80219a468dffcf0ffe94abfR2389

File: inc/pdu_rack.class.php [hunk 6]
```diff
@@ -715,18 +715,17 @@ static function getUsed() {
     */
    static function getOtherSide($side) {
       switch ($side) {
-         case self::SIDE_TOP;
+         case self::SIDE_TOP:
             return self::SIDE_BOTTOM;
-            break;
-         case self::SIDE_BOTTOM;
+         case self::SIDE_BOTTOM:
```
Comment (cedric-anne) [inline inc/pdu_rack.class.php:720]:
```suggestion
            break;
```

File: inc/pdu_rack.class.php [hunk 7]
```diff
@@ -715,18 +715,17 @@ static function getUsed() {
...
+         case self::SIDE_BOTTOM:
             return self::SIDE_TOP;
-            break;
-         case self::SIDE_LEFT;
+         case self::SIDE_LEFT:
```
Comment (cedric-anne) [inline inc/pdu_rack.class.php:723]:
```suggestion
            break;
```

File: inc/pdu_rack.class.php [hunk 8]
```diff
@@ -715,18 +715,17 @@ static function getUsed() {
...
+         case self::SIDE_LEFT:
             return self::SIDE_RIGHT;
-            break;
-         case self::SIDE_RIGHT;
+         case self::SIDE_RIGHT:
+         case self::SIDE_RIGHT:
             return self::SIDE_LEFT;
```
Comment (cedric-anne) [inline inc/pdu_rack.class.php:728]: IMHO, we should keep breaks even after returns. If someone decide to replace return instructions by something else, he may forgot to add the break.
```suggestion
            return self::SIDE_LEFT;
            break;
```

File: inc/search.class.php (hunk 3 above)
Comment (cedric-anne) [reply inc/search.class.php:6129]: `break` is OK here.

There was previously a `return` instruction that has been removed, so code was never made to continue in next case.

See https://github.com/glpi-project/glpi/commit/05be96466f63dfd68ab100f37a3b8791389053de

File: inc/user.class.php [hunk 9]
```diff
@@ -3799,6 +3799,7 @@ static function getSqlSearchResult ($count = true, $right = "all", $entity_restr
                            'glpi_profilerights.rights'   => ['&', KnowbaseItem::READFAQ]
                         ] + getEntitiesRestrictCriteria('glpi_profiles_users', '', $entity_restrict, 1)
                      ];
+                     break;
```
Comment (cedric-anne) [inline inc/user.class.php:3802]: According description of #561 :roll_eyes: , I guess the lack of `break` was just a mistake. So OK for this one.

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**